crates/but-ctx/src/legacy.rs

```diff
@@ -63,6 +63,15 @@ impl Context {
     }
 }
 
+impl TryFrom<LegacyProjectId> for Context {
+    type Error = anyhow::Error;
+
+    fn try_from(value: LegacyProjectId) -> Result<Self, Self::Error> {
+        let project = gitbutler_project::get(value)?;
+        Context::new_from_legacy_project(project)
```

Comment:
Since this gets executed from within a macro... in theory error could be extra confusing. I wonder, perhaps this is a place we could be helpful when debugging if we included some additional error info with the anyhow's `.context`? What do you think?